emcc.py

```diff
@@ -1466,6 +1466,9 @@ def is_supported_link_flag(f):
       # To ensure allocated thread stacks are aligned:
       shared.Settings.EXPORTED_FUNCTIONS += ['_memalign']
 
+      if shared.Settings.PROXY_TO_PTHREAD:
+        shared.Settings.EXPORTED_FUNCTIONS += ['_proxy_main']
```

Comment:
Great, this is certainly better - not sure why this didn't cross my mind when originally writing this :)